interfaces/builtin/uhid_test.go

```diff
@@ -0,0 +1,149 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/snaptest"
+)
+
+type UhidInterfaceSuite struct {
+	iface interfaces.Interface
+
+	testSlot1 *interfaces.Slot
+
+	testUdevBadValue1     *interfaces.Slot
+	testUdevBadValue2     *interfaces.Slot
+	testUdevBadValue3     *interfaces.Slot
+	testUdevBadValue4     *interfaces.Slot
+	testUdevBadInterface1 *interfaces.Slot
+
+	plug *interfaces.Plug
+}
+
+var _ = Suite(&UhidInterfaceSuite{
+	iface: &builtin.UhidInterface{},
+})
+
+func (s *UhidInterfaceSuite) SetUpTest(c *C) {
+	// Mocking
+	osSnapInfo := snaptest.MockInfo(c, `
+name: ubuntu-core
+type: os
+slots:
+  test-slot-1:
+    interface: uhid
+    path: /dev/uhid
+  test-udev-bad-value-1:
+    interface: uhid
+    path: /dev/i2c-1
+  test-udev-bad-value-2:
+    interface: uhid
+    path: ""
+  test-udev-bad-value-3:
+    interface: uhid
+    path: /dev/uhid-1
+  test-udev-bad-value-4:
+    interface: uhid
+    path: /uhid
+  test-udev-bad-interface-1:
+    interface: other-interface
+    path: /dev/uhid
```

Comment:
I'm confused about all these new 'path' attributes. Before the interface was only about accessing /dev/uhid. Now you added a path attribute for /dev/uhid-* and bad attributes. Was this just from trying to rework iio for this? If so, just get rid of all these "path" tests, the "path" attribute in the SanitizeSlot() and the "path" in ConnectedPlugSnippet. All I meant for you to do in ConnectedPlugSnippet is something like:

```
	case interfaces.SecurityUDev:
		var tagSnippet bytes.Buffer
		const path = "/dev/uhid"
		const udevRule = `KERNEL=="%s", TAG+="%s"`
		for appName := range plug.Apps {
			tag := udevSnapSecurityName(plug.Snap.Name(), appName)
			tagSnippet.WriteString(fmt.Sprintf(udevRule, path, tag))
			tagSnippet.WriteString("\n")
		}
		return tagSnippet.Bytes(), nil
	}
```